plugins/woocommerce/includes/class-wc-order.php

```diff
@@ -14,6 +14,91 @@
  * These are regular WooCommerce orders, which extend the abstract order class.
  */
 class WC_Order extends WC_Abstract_Order {
+	/**
+	 * The following constants are used to define the order status.
+	 * For a full documentation on the public order statuses, please refer to the following link:
+	 * https://woocommerce.com/document/managing-orders/order-statuses/
+	 */
+
+	/**
+	 * The order has been received, but no payment has been made.
+	 *
+	 * @var string
+	 */
+	const STATUS_PENDING = 'pending';
+
+	/**
+	 * The customer’s payment failed or was declined, and no payment has been successfully made.
+	 *
+	 * @var string
+	 */
+	const STATUS_FAILED = 'failed';
+
+	/**
+	 * The order is awaiting payment confirmation.
+	 *
+	 * @var string
+	 */
+	const STATUS_ON_HOLD = 'on-hold';
+
+	/**
+	 * Order fulfilled and complete.
+	 *
+	 * @var string
+	 */
+	const STATUS_COMPLETED = 'completed';
+
+	/**
+	 * Payment has been received (paid), and the stock has been reduced.
+	 *
+	 * @var string
+	 */
+	const STATUS_PROCESSING = 'processing';
+
+	/**
+	 * Orders are automatically put in the Refunded status when an admin or shop manager has fully refunded the order’s value after payment.
+	 *
+	 * @var string
+	 */
+	const STATUS_REFUNDED = 'refunded';
+
+	/**
+	 * The order was canceled by an admin or the customer.
+	 *
+	 * @var string
+	 */
+	const STATUS_CANCELLED  = 'cancelled';
+	const STATUS_TRASH      = 'trash';
+	const STATUS_AUTO_DRAFT = 'auto-draft';
+
+	/**
+	 * Draft orders are created when customers start the checkout process while the block version of the checkout is in place.
+	 *
+	 * @var string
+	 */
+	const STATUS_DRAFT = 'draft';
+
+	/**
+	 * Order statuses that are for internal use only.
+	 *
+	 * @var array
+	 */
+	const INTERNAL_STATUSES = array(
```

Comment:
It seems `INTERNAL_STATUSES` and `DRAFT_STATUSES` are not used in this PR, are they needed for the followup PRs?

Reply:
Somewhat related, if we keep this should we add an entry for `'checkout-draft'` (or else make that a standalone constant of its own)? Same question for `'new'`.

(Asking because of [this line](https://github.com/woocommerce/woocommerce/blob/9.3.3/plugins/woocommerce/includes/data-stores/class-wc-order-data-store-cpt.php#L208), though I don't know its history or when exactly these come into play.)

Reply:
The idea for both was to be used for documentation purposes only and maybe when those are grouped in the code. But I think it makes more sense to wait for that "need" before introducing them.

I have also introduced the `new` status constant so we can replace that reference on the next PRs.